src/pool.rs

```diff
@@ -0,0 +1,104 @@
+use crate::client::Client;
+use crate::cmd::{Command, Get, Set};
+use crate::Result;
+use bytes::Bytes;
+use std::time::Duration;
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender};
+use tokio::sync::oneshot::{channel, Sender as OneshotSender};
```

Comment:
I believe it is more idiomatic to do:

```suggestion
use tokio::sync::oneshot::{self, channel};
```

Then, the `Sender` is references as:

```rust
```